sdks/java/io/tika/src/test/java/org/apache/beam/sdk/io/tika/ParseResultTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.tika;
+
+import static org.junit.Assert.assertEquals;
+import org.apache.tika.metadata.Metadata;
+import org.junit.Test;
+
+/**
+ * Tests ParseResult.
+ */
+public class ParseResultTest {
+  @Test
+  public void testEqualsAndHashCode() {
```

Comment:
No Reply option is available under the TikaOptions comment since I'm replying here, I thought shipping TikaOptions would let every user avoid create their own TikaOptions capturing the command line parameters ? Do you reckon it can still make sense to keep it, I believe the number of options which the users would want to pass from the cmd will grow ?

Reply:
I don't see why a user would create a `TikaOptions` at all in their program: in this respect `TikaIO` is not different from TextIO or AvroIO etc., and I have yet to see a pipeline that has a user define a `TextOptions` or `AvroOptions`.

`PipelineOptions` are typically used only by the pipeline author themselves, to extract individual options specific to their particular pipeline and configure various transforms using them (e.g. a user might have a pipeline with an option called `--clicksTableName` and pass it to `BigQueryIO.write().to()` to write click events). Almost no transforms use `PipelineOptions` directly; perhaps, this should be forbidden. `PipelineOptions` is also global to the entire pipeline, so if you have 3 `TikaIO` transforms created using `fromOptions`, they'll be configured identically, which is likely not useful.

Reply:
OK, I can move it to a small sample application of mine